### DISCUSSION TITLE
[Feature request] override caption styles (font, size, color, position)

### DISCUSSION THREAD

**Discussion by irasantiago** [vimeo/player.js#1010] [Feature requests]
Hi there, 

It would be great if there were additional options that allowed for:
- Updating the font and font-size. 
  - Example use case: adding a custom control to adjust the text size of the captions.
- Changing the color of the text and background. 
  - Example use case: help support high contrast mode for accessibility, set colors to improve contrast based on video content.
- Some limited options for positioning. 
  - Example use case: setting a max-width for captions on a cropped video, reposition captions in videos that include text. 

# Related issues

 #267 
 #616

**Comment by Tumbuctoo** [comment]
Completely agree! 
Vimeo has this strange behavior with subtitles on Plyr's player. Something should be done from Vimeo's side.

Vimeo would also do us all a big service to implement the possibility to convert plain .srt subtitles to WebVTT subtitles which could then be edited / stylized  directly into Vimeo.

**Comment by fisherinnovation** [comment]
Thanks a ton for your feedback! We will keep this ticket open and provide more updates when they become available.

**Comment by Philippe-MTL** [comment]
Seems that subtitle customization has been implemented on Vimeo recently.
Sadly, color tags in SRT or VTT files are now ignored...

![Clipboard01](https://user-images.githubusercontent.com/60315564/168936095-7f2bee6c-5bae-47b9-8793-4338815e6c5c.jpg)

**Comment by Tumbuctoo** [comment]
Merci beaucoup pour l'info!

**Comment by steve52** [comment]
Hi, @Philippe-MTL ! Vimeo dev here. I wasn’t aware of any color tags working previously in the vimeo player. Can you give an example of what sort of tags you were able to include previously that changed the color of captions?

**Comment by Philippe-MTL** [comment]
Hi Steve,

I used to give the song title in blue, to differentiate it from the text of
the song, as in :

WEBVTT

00:00:00.030 --> 00:00:03.062
*<c.cyan>*Erik Satie - Ludions (Léon-Paul Fargue)
1. Air du rat - The Rat's Air*</c>*
Abi Abirounère

00:00:03.617 --> 00:00:05.937
Qui que tu n'étais don?
who then that you were not?

this doesn't work anymore as Vimeo's subtitle customization is applied to
the whole file...

Thanks for looking,


- Philippe

**Comment by steve52** [comment]
So that should be adding a class of "cyan" to that block of text, but we were never actually styling it in the player. Maybe the site which the video is on was in some way and now the new styles are overriding it. Do you have a link to the site that this video is on? I'd be curious to see.

**Comment by Philippe-MTL** [comment]
Here we go :

https://vimeo.com/manage/videos/684278688/3e7a52bd5e

Unfortunately I could not find any screenshot to document it...

- Philippe

**Comment by a10tive1** [comment]
POSITIONING via an uploaded WebVTT file does not work either. This will be a big problem for me, as I often have to reposition specific captions so they don't overlay onto existing lower thirds. Has anyone figured out a hack to override Vimeo's fixed caption positioning?

**Comment by rboixaderg** [comment]
Any news about positioning via an uploaded WebVTT file? Is there another way to modify the position of the caption?

Thanks

**Comment by Wesleyzxc** [comment]
Do we have any updates on this? I'm trying to add styling to an uploaded VTT file and I see it processed but not applied.

**Comment by rkrishnan8594** [comment]
Hey all - while adding support for styled WebVTT files is a product goal of ours, it's not on our roadmap for the next few months. It's most likely that we add support some time in 2024. I'll make sure we provide updates to this issue when there's news to share.

**Reply by illtrax** [reply]
Any chance we can get an update? Would be great to be able to move captions off lower thirds. Cheers!

**Comment by gradosevic** [comment]
Hi folks, 

I’ve created a PR that adds a **safe placeholder** for caption styling support:

[https://github.com/vimeo/player.js/pull/1104](https://github.com/vimeo/player.js/pull/1104)

The goal is to let developers read and set subtitle style via the Player API, so we can persist user preferences (font size, color, background, etc.) across videos on our end (using localStorage or cookies).

Vimeo wouldn't need to store or manage anything — just expose the current styles and optionally emit changes via `postMessage`.

If this is something you’d also benefit from, feel free to chime in or 👍 the PR to help it get visibility!
